google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/facet/impl/AppEngineJavaeeSupportContributor.java

```diff
@@ -68,7 +68,7 @@ private void setWebXmlServletVersion(Module module) {
     StartupManager.getInstance(module.getProject()).runWhenProjectIsInitialized(() -> {
       for (WebFacet webFacet : WebFacet.getInstances(module)) {
         final WebApp webApp = webFacet.getRoot();
-        if (webApp == null) {
+        if (webApp == null || !webApp.getXmlTag().isWritable()) {
```

Comment:
Also, I know it wasn't there before, but since we do have a test for this class would it be possible to add some coverage to assert that things work properly regarding the writable status?